tf-psa-crypto/drivers/builtin/src/ecp.c

```diff
@@ -2914,6 +2914,9 @@ int mbedtls_ecp_check_privkey(const mbedtls_ecp_group *grp,
         if (mbedtls_mpi_get_bit(d, 0) != 0 ||
             mbedtls_mpi_get_bit(d, 1) != 0 ||
             mbedtls_mpi_bitlen(d) - 1 != grp->nbits) {  /* mbedtls_mpi_bitlen is one-based! */
+                                                        /* Zero limb mpis are invalid, but even if
+                                                           one is provided for d, the grp->nbits
+                                                           can not be equal to UINT64_MAX. */
```

Comment:
Sorry, this comment is wrong.

> Zero limb mpis are invalid

1. Zero-limb `mbedtls_mpi` are perfectly valid to represent the value 0.
2. Zero-limb MPIs are not relevant here. `mbedtls_mpi_bitlen(d)` is 0 when the value of `d` is 0, regardless of how many limbs are used to represent it.

> can not be equal to `UINT64_MAX`

`UINT64_MAX` is irrelevant here, you meant `SIZE_MAX`. This computation doesn't involve `uint64_t` values. Both `mbedtls_mpi_bitlen(d)` and `grp->nbits` are `size_t` values.